conformance/05_referrers.go

```diff
@@ -0,0 +1,301 @@
+					BeNumerically("<", 300)))
+				Expect(resp.Header().Get("OCI-Subject")).To(Equal(manifests[4].Digest))
+			})
+
+			g.Specify("Populate registry with test references manifest (ArtifactType, config.MediaType = emptyJSON)", func() {
+				SkipIfDisabled(referrers)
+				RunOnlyIf(runReferencesSetup)
+				req := client.NewRequest(reggie.PUT, "/v2/<name>/manifests/<reference>",
+					reggie.WithReference(refsManifestBLayerArtifactDigest)).
+					SetHeader("Content-Type", "application/vnd.oci.image.manifest.v1+json").
+					SetBody(refsManifestBLayerArtifactContent)
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAll(
+					BeNumerically(">=", 200),
+					BeNumerically("<", 300)))
+				Expect(resp.Header().Get("OCI-Subject")).To(Equal(manifests[4].Digest))
+			})
+		})
+
+		g.Context("Get references", func() {
+			g.Specify("GET request to nonexistent blob should result in empty 200 response", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(dummyDigest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).To(BeZero())
+			})
+
+			g.Specify("GET request to existing blob should yield 200", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(manifests[4].Digest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if h := resp.Header().Get("Docker-Content-Digest"); h != "" {
+					Expect(h).To(Equal(configs[4].Digest))
+				}
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).To(Equal(4))
+				Expect(index.Manifests[0].Digest).ToNot(Equal(index.Manifests[1].Digest))
+			})
+
+			g.Specify("GET request to existing blob with filter should yield 200", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(manifests[4].Digest)).
+					SetQueryParam("artifactType", testRefArtifactTypeA)
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if h := resp.Header().Get("Docker-Content-Digest"); h != "" {
+					Expect(h).To(Equal(configs[4].Digest))
+				}
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).To(Equal(2))
+
+				// also check resp header "OCI-Filters-Applied: artifactType" denoting that an artifactType filter was applied
+				Expect(resp.Header().Get("OCI-Filters-Applied")).To(EqualOrWarn(testRefArtifactTypeA))
```

Comment:
Since filter support is a "SHOULD",  the length and header check here needs to be conditional. If the header is set, then we expect 2 entries and the header must match the expected value. Else, we'd expect 4 manifests.

Reply:
Fixed.